sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -230,10 +232,13 @@ case class PreprocessTableCreation(catalog: SessionCatalog) extends Rule[Logical
         DDLUtils.checkTableColumns(tableDesc.copy(schema = analyzedQuery.schema))
 
         val output = analyzedQuery.output
-        val partitionAttrs = normalizedTable.partitionColumnNames.map { partCol =>
-          output.find(_.name == partCol).get
-        }
-        val newOutput = output.filterNot(partitionAttrs.contains) ++ partitionAttrs
+        val partitionColumnNames = HashSet(normalizedTable.partitionColumnNames: _*)
+        // testing...
+        val newOutput = output.filterNot(
```

Comment:
use `output.partition`?

Reply:
Turnes out we need the order of partition attributes to be the same as `normalizedTable.partitionColumnNames` (caught it in the UTs). Fixed